### ISSUE TITLE
Improved vertex color functionality

### ISSUE THREAD

**Issue by amplified1** [Yellow-Dog-Man/Resonite-Issues#678]
### Is your feature request related to a problem? Please describe.

Resonite is currently very rigid and limited with its utilization of vertex colors. You can only use the vertex colors of a mesh for one channel of a material and there is only one vertex color channel available for meshes, despite multiple export formats supporting multiple vertex color channels. EG: GLTF can store vertex colors as an array, with multiple colors per vertex.

### Describe the solution you'd like

There could be a universal toggle on every texture input to use vertex colors, and an input for vertex color channel index. When shader ProtoFlux is also eventually implemented after the implementation of the new rendering engine there should also be a vertex color node that has an option for channel index. Any future mesh editing options should also have the ability to modify vertex colors on multiple channels.

### Describe alternatives you've considered

N/A

### Additional Context

This probably shouldn't be considered until the new renderer is implemented, but I feel like it should be kept in mind as it would be very useful, allowing for users to make sophisticated shading setups using only vertex colors, or a combination of vertex colors and textures.

**Comment by Frooxius** [comment]
This will mostly have to wait for custom shader support, so you can access and utilize the vertex color data in ways you want.

Since the shaders are currently just a set we provide, you'll get the combinations and uses we have made. We could potentially add a few more, but that's always going to be limited.

**Comment by Comptr-user** [comment]
Would this feature request cover "Ability to give a mesh vertex colors from a texture," or should I make a separate issue for that?

**Comment by Frooxius** [comment]
That sounds like a different function to me, so it'd be separate request.

The issue is about shaders using existing vertex colors when rendering. Your request sounds like baking texture colors into vertex data?

**Comment by Comptr-user** [comment]
Yes, my request would be baking textures into vertex data.
Not sure if this is the right place to get into it, but a small project I was working on would be much simpler/easier if the avatar I'm using had vertex colors. Blender could be used to apply vertex colors, but I don't have the source file for the model so that's not an option for me.

I was asking about if that idea would fit under this pull request because the line "[...] universal toggle on every texture input to use vertex colors, and an input for vertex color channel index" sounded vaguely like baking textures into vertex colors, and I didn't want to make a duplicate/redundant issue if that feature would already be covered here.

**Comment by Frooxius** [comment]
No, that'd be different request from this. I'd recommend making a separate issue for that please!
